### PR TITLE
modularization of observation space

### PR DESCRIPTION
**Pull request by luigiberducci**
Addresses #71. 
It keeps compatibility with old observation space and implements dictionary-based observations for "kinematic_state", "dynamic_state" and arbitrary "features"

### CODE REVIEW

File: gym/f110_gym/envs/f110_env.py [hunk 1]
```diff
@@ -156,6 +158,13 @@ def __init__(self, render_mode=None, **kwargs):
         except:
             self.integrator = Integrator.RK4
 
+        # observation
```
Comment (hzheng40) [inline gym/f110_gym/envs/f110_env.py:161]: This should probably follow the new config dict we'll be using.
We'll fix after merging the other pr.

**Review by hzheng40 — CHANGES_REQUESTED** [review]
Setting config for observation type should follow the proposed change in #74

File: gym/f110_gym/envs/base_classes.py [hunk 2]
```diff
@@ -561,7 +565,7 @@ def step(self, control_inputs):
         for i, agent in enumerate(self.agents):
             # update each agent's pose
             current_scan = agent.update_pose(control_inputs[i, 0], control_inputs[i, 1])
-            agent_scans.append(current_scan)
+            self.agent_scans[i, :] = current_scan
```
Comment (nandantumu) [inline gym/f110_gym/envs/base_classes.py:568]: This is a good catch, I would expect this to result in a speedup

**Review by nandantumu — APPROVED** [review]
LGTM

**Comment by luigiberducci** [comment]
Using the new config dictionary now. 
Example script modified to use `kinematic_state` instead of original observation.

@hzheng40 @nandantumu please check the changes

**Comment by hzheng40** [comment]
Looks good, merging.

**Review by hzheng40 — APPROVED** [review]

**Pull request merged**